test/test_reason.re

```diff
@@ -1093,6 +1150,126 @@ let testWithId = () => {
   });
 };
 
+let testPropMaybeNone = () =>
+  withContainer(c => {
+    act(() =>
+      React.Dom.render(<div className=?None />, Dom_html.element(c))
+    );
+    assert_equal(c##.innerHTML, Js.string("<div></div>"));
+  });
+
+let testPropMaybeSome = () =>
+  withContainer(c => {
+    act(() =>
+      React.Dom.render(
+        <div className=?{Some("foo")} />,
+        Dom_html.element(c),
+      )
+    );
+    assert_equal(c##.innerHTML, Js.string("<div class=\"foo\"></div>"));
+  });
+
+let testPropCustomString = () =>
+  withContainer(c => {
+    module Prop = {
+      let foo = Prop.string("foo");
+    };
+    act(() => React.Dom.render(<div foo="bar" />, Dom_html.element(c)));
```

Comment:
It is possible with some overrides to create invalid elements, but I don't think it's a big issue.